subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/api/ConfigurationDefaultsIntegrationTest.groovy

```diff
@@ -308,4 +308,24 @@ task check {
         expect:
         succeeds ":check"
     }
+
+    def "copied configuration have unique names"() {
+        buildFile << """
+            configurations {
+              conf
+            }
+
+            task check {
+                doLast {
+                    assert configurations.conf.copyRecursive().name == 'confCopy'
```

Comment:
should the first one have the `Copy` suffix as well?

Reply:
I intentional left the _first_ copy out here to keep the potential breaking change minimal. If you just create one copy of a config and relate on the existing pattern of `origin` + `Copy` this does not break your code. Instead only further copies applied that suffix (When doing multiple copies imo the likelyhood of your code relying on the name seems lower then)

Reply:
Got it, thanks!